include/boost/date_time/posix_time/time_serialize.hpp

```diff
@@ -11,29 +11,45 @@
 
 #include "boost/date_time/posix_time/posix_time.hpp"
 #include "boost/date_time/gregorian/greg_serialize.hpp"
+#include "boost/core/nvp.hpp"
 #include "boost/numeric/conversion/cast.hpp"
-#include "boost/serialization/split_free.hpp"
-#include "boost/serialization/nvp.hpp"
-#include "boost/serialization/version.hpp"
-
-// macros to split serialize functions into save & load functions
-// NOTE: these macros define template functions in the boost::serialization namespace.
-// They must be expanded *outside* of any namespace
-BOOST_SERIALIZATION_SPLIT_FREE(boost::posix_time::ptime)
-BOOST_SERIALIZATION_SPLIT_FREE(boost::posix_time::time_duration)
-BOOST_SERIALIZATION_SPLIT_FREE(boost::posix_time::time_period)
+#include "boost/type_traits/integral_constant.hpp"
 
 // Define versions for serialization compatibility
 // alows the unit tests to make an older version to check compatibility
 #ifndef BOOST_DATE_TIME_POSIX_TIME_DURATION_VERSION
 #define BOOST_DATE_TIME_POSIX_TIME_DURATION_VERSION 1
 #endif
 
-BOOST_CLASS_VERSION(boost::posix_time::time_duration, BOOST_DATE_TIME_POSIX_TIME_DURATION_VERSION)
-
 namespace boost {
 namespace serialization {
 
+template<typename T>
+struct version;
+
+template<>
+struct version<boost::posix_time::time_duration>
```

Comment:
Is this what serialization requires for this?  Can it not just be a const static?

Reply:
Initially I have done exactly that, and it is what documentations says, but in reality Serialization ODR-uses the value and you get linking error, to fix it you need an out of class definition, so it is simpler to inherit from `integral_constant`.

Reply:
right, but it requires another header dependency -- which of course in c++11 should really be std:: instead of boost.  and maybe it should even just be constexpr which I think should avoid the linking issue.

Reply:
> right, but it requires another header dependency 

The other DateTime parts are implicitly include it already anyway

> and maybe it should even just be constexpr which I think should avoid the linking issue.

AFAIK only `inline` keyword will solve the ODR violation, but it requires C++17.

Reply:
> The other DateTime parts are implicitly include it already anyway

Ok, that's fair.

> AFAIK only inline keyword will solve the ODR violation, but it requires C++17.

yep, that's true.